source/initial_conditions/adiabatic_boundary.cc

```diff
@@ -0,0 +1,277 @@
+      const double lat_degrees = lat * (180 / numbers::PI);
+
+      /* Set all longitudes between [0,360]. */
+      if (lon_degrees < 0)
+        return std::make_pair(lat_degrees, lon_degrees + 360);
+      else if (lon_degrees > 360)
+        return std::make_pair(lat_degrees, lon_degrees - 360);
+      else
+        return std::make_pair(lat_degrees, lon_degrees);
+
+    }
+
+    template <>
+    double
+    AdiabaticBoundary<2>::get_isotherm_depth(const double,
+                                             const double) const
+    {
+      AssertThrow(false, ExcInternalError());
+      return 0;
+    }
+
+    template <int dim>
+    double
+    AdiabaticBoundary<dim>::get_isotherm_depth(const double latitude,
+                                               const double longitude) const
+    {
+      /**
+       * Loop over the entire array and see if we find a point
+       * that's within delta of what we're looking for. If this
+       * point is not within the delta then test if it is within twice
+       * as much as delta.
+       */
+      for (unsigned int i = 0; i <= depths_iso.size();)
+        if (std::fabs(latitude - latitudes_iso[i]) <= delta && std::fabs(longitude - longitudes_iso[i]) < delta)
+          {
+            return -depths_iso[i]*1000;
+          }
+        else if (std::fabs(latitude - latitudes_iso[i]) <= delta*2 && std::fabs(longitude - longitudes_iso[i]) < delta*2)
+          {
+            return -depths_iso[i]*1000;
+          }
+        else
+          i++;
+      Assert(false, ExcInternalError());
+      return 0;
+    }
+
+    /**
+     * Calculate distance from the center of the Earth to the surface of WGS84.
+     */
+
+    template <int dim>
+    double
+    AdiabaticBoundary<dim>::radius_WGS84(const double &theta) const
+    {
+      const double eccentricity    = 8.1819190842622e-2;
+      const double semi_major_axis = 6378137.0;
+      return semi_major_axis/std::sqrt(1- eccentricity * eccentricity
+                                       * std::sin(numbers::PI*theta/180)*std::sin(numbers::PI*theta/180));
+    }
+
+    template <int dim>
+    double
+    AdiabaticBoundary<dim>::initial_temperature (const Point<dim> &) const
+    {
+      Assert (false, ExcNotImplemented());
+      return 0;
+    }
+
+    /**
+     * Get the depth of the adiabatic boundary isotherm for the current lat/long.
+     * If above the isotherm use a linear interpolation to the surface.
+     * If below the isotherm use an increase of .5 degrees per kilometer.
+     */
+
+    template <>
+    double
+    AdiabaticBoundary<3>::initial_temperature (const Point<3> &position) const
+    {
+      const std::pair<double, double> lat_long = lat_long_from_xyz_WGS84(position);
+      const double depth                       = position.norm() - radius_WGS84(lat_long.first);
+      const double isotherm_depth              = get_isotherm_depth(lat_long.first, lat_long.second);
+      if (depth < isotherm_depth)
```

Comment:
Due to the sign issue mentioned above this logic seems very strange to me (although likely correct). You ask here if the depth of this point is smaller than the isotherm_depth, so I first think: Ok, this is the lithosphere. But from the calculation below I figure out that this must be the mantle part (because the surface_temperature is used in the `else` branch). Then I think: Ok this is the mantle part, but in the calculation below you subtract something from the isotherm_temperature, which is counter-intuitive, because the temperature should be rising. Finally, I see that you subtract something negative (because depth < isotherm_depth) as in the if() condition.
Please do not get me wrong, this code is correct, and I am not picky, because I want you to have extra work, I am just trying to understand the code, because I know that I (and many others) will have to understand it again and again while maintaining it, so making it easier to understand is very important (also to you, if you read it again in half a year :wink:).